python/arcticdb/diagnostics_engine/library_options.py

```diff
@@ -0,0 +1,286 @@
+    \n- Additionally, if a Pandas DataFrame contains even a single non-primitive data type object (i.e., not an
+    integer, float, string, or boolean), the entire DataFrame will need to be pickled. This could potentially
+    impact performance and the ease of data manipulation.
+    
+    \n For a more detailed understanding of pickling, including how it works and its various use cases and
+    limitations, please refer to the specific 'Pickling' section in this notebook.
+    """,
+    "symbol_list": """
+    \n If enabled, the symbol list caching functionality of Arctic will be used. This is designed to improve
+    performance of list_symbols operations. For a more detailed understanding of this feature, including how it
+    works and its various use cases and limitations, please refer to the specific 'Symbol list consistency
+    diagnosis' section in this notebook.""",
+}
+
+
+class LibraryInfoGetter:
+    """
+    This class provides a method for displaying and explaining the various options
+    associated with a specific library. It aims to provide users with a clear understanding
+    of each option, including its purpose, usage implications, and potential impact on
+    library operations.
+
+    Methods:
+    --------
+    display_library_options():
+        Fetches and displays the options for a given library, along with detailed
+        explanations for each option.
+
+    Example Usage:
+    --------------
+    info_getter = LibraryInfoGetter(lib, EXPLANATIONS)
+    info_getter.display_library_options()
+    """
+
+    def __init__(self, lib, explanations):
+        self.response_obj = lib._lib_cfg.lib_desc.version
+        self.explanations = explanations
+
+        self.response_non_default_str = (
+            "## Next, see the list of your library options that has been modified with respect to the default"
+            " values:\n\n"
+        )
+        self.response_default_str = (
+            "## Next, see the list of your library options that remains with the default values:\n\n"
+        )
+
+        # Default values defined internally
+        self.internal_default_values = {
+            "column_group_size": 127,
+            "segment_row_size": 100000,
+            "max_num_buckets": 150,
+        }
+
+    def traverse_descriptor(self, obj):
+        """
+        Traverses the fields of the provided protobuf object, compares their values with default ones,
+        and updates response strings accordingly. The function is recursive for nested protobuf message types.
+        """
+        default_response_obj = type(obj)()
+        for field in obj.DESCRIPTOR.fields:
+            value = getattr(obj, field.name)
+            if field.message_type:
+                self.traverse_descriptor(value)
+            elif field.name in self.explanations:
+                default_value = getattr(default_response_obj, field.name)
+                explanation = self.explanations.get(field.name, "No explanation required.")
```

Comment:
I am unsure of what do you exactly mean in your comment:

1. `default_value` is used in line 240, in order to determine whether we changed the original value or not.
2. I intentionally want to control what options to show to the user. It is why `field.name` must be included in `self.explanations`.

Reply:
1. I meant the default value which is the second argument to `get`
2. We should at least print that we don't have a detailed explanation for the setting in this case

Reply:
ok. I see your point. That's certainly confusing because the existing logic makes the "No explanation required." default message totally unnecessary. It was intentional not to show options that we don't provide explanations. But happy to show them anyway.

Reply:
Created an exclusion_set because there are some protobuf fields that are not considered optional per se when they're traversed recursively.